helm_deploy/payforlegalaid-ui/templates/deployment.yaml

```
@@ -0,0 +1,80 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "payforlegalaid-ui.fullname" . }}
+  labels:
+    {{- include "payforlegalaid-ui.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "payforlegalaid-ui.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "payforlegalaid-ui.labels" . | nindent 8 }}
+          {{- with .Values.podLabels }}
+          {{- toYaml . | nindent 8 }}
+          {{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "payforlegalaid-ui.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.service.port }}
+              protocol: TCP
+          livenessProbe:
```

Is this necessary? Looks scope creep to me

---

I appreciate you flagging this - it's important we stay focused on priorities. The probes might seem like scope creep, but they're actually critical to our deployment reliability.

Here's why:

Without probes, we're flying blind:
- Kubernetes will assume the pod is healthy the moment it starts, even if the app crashes immediately after
- We'd see 'running' pods that can't actually serve requests, creating silent failures

This prevents production issues: like an incident where a service deployed 'successfully' but was stuck in a broken state
- Probes would have detected this and triggered auto-recovery
- Minimal maintenance overhead:
- The probe configuration is a one-time setup